packages/block-editor/src/components/block-list/block.native.js

```diff
@@ -337,7 +340,9 @@ export default compose( [
 			! isDescendantSelected &&
 			( isDescendantOfParentSelected || rootBlockId === clientId );
 
-		const isInnerBlockHolder = name === getGroupingBlockName();
+		// TODO: find better way to handle full border for columns ( maybe return array from getGoupingBlockName )
+		const isInnerBlockHolder =
+			name === getGroupingBlockName() || name === 'core/columns';
```

Comment:
Here's the place that I also feel we need to replace `name === 'core/columns'` with something more generic